test/1_stdlib/Optional.swift

```diff
@@ -16,211 +21,133 @@ extension ImplicitlyUnwrappedOptional where Wrapped : TestProtocol1 {
-if x != nil { 
-  print("x is non-empty!")
-} else { 
-  print("an empty optional is logically false")
-}
-// CHECK: an empty optional is logically false
-
-switch x {
-case .Some(let y):
-  assert(false, "Something's wrong here!")
-case .None:
-  ()
-}
+OptionalTests.test("nil comparison") {
+  var x: Int? = nil
+  expectFalse(x != nil)
 
-x = .Some(0)
-
-x = .Some(1)
+  switch x {
+  case .Some(let y): expectUnreachable()
+  case .None: break
+  }
 
-if x != nil {
-  print("a non-empty optional is logically true") 
-} else { 
-  assert(false, "x is empty!")
-}
-// CHECK: a non-empty optional is logically true
+  x = .Some(1)
+  expectTrue(x != nil)
 
-if x == nil { 
-  print("logical negation fails 0")
-} else { 
-  print("logical negation works 0") 
-}
-// CHECK: logical negation works 0
+  if true {
+    var y1: Int? = .None
+    expectTrue(y1 == nil)
 
-if true {
-  var y1 : Optional<Int> = .None
-  if y1 == nil {
-    print("y1 is .None")
+    var y2: Int? = .None
+    expectTrue(y2 == nil)
   }
-  // CHECK: y1 is .None
 
-  var y2 : Optional<Int> = .None
-  if y2 == nil {
-    print("y2 is .None")
-  }
-  // CHECK: y2 is .None
-}
+  let x1: Int? = nil
+  let x2: Int? = .None
 
-func optional_param(x: Optional<Int>) {
-  if x == nil {
-    print("optional param OK")
-  }
-}
-optional_param(.None)
-// CHECK: optional param OK
+  expectTrue(x1 == nil)
+  expectTrue(x2 == nil)
 
-func optional_return() -> Optional<Int> {
-  return .None
-}
-if optional_return() == nil {
-  print("optional return OK")
-}
-// CHECK: optional return OK
+  switch x {
+    case .Some(let y): expectEqual("\(y)", "1")
+    case .None: assert(false)
+  }
 
-switch x {
-case .Some(let y):
-  print("destructuring bind: \(y).")
-case .None:
-  ()
+  expectEqual("forced extraction: \(x!).", "forced extraction: 1.")
+  expectEqual("forced extraction use: \(x!.successor()).", "forced extraction use: 2.")
 }
-// CHECK: destructuring bind: 1.
-
-
-print("forced extraction: \(x!).")
-// CHECK: forced extraction: 1.
 
-print("forced extraction use: \(x!.successor()).")
-// CHECK-NEXT: forced extraction use: 2.
-
-func testRelation(p: (Int?, Int?) -> Bool) {
+func testRelation(p: (Int?, Int?) -> Bool) -> [Bool] {
   typealias optPair = (Int?, Int?)
   
   let relationships: [optPair] = [
     (1, 1), (1, 2), (2, 1), (1, .None), (.None, 1), (.None, .None)
   ]
 
-  var prefix = ""
-  for (l,r) in relationships {
-    print("\(prefix)\(p(l, r))", terminator: "")
-    prefix=", "
-  }
-  print(".")
+  return relationships.map { p($0, $1) }
 }
 
-testRelation(==)
-// CHECK-NEXT: true, false, false, false, false, true.
-
-testRelation(!=)
-// CHECK-NEXT: false, true, true, true, true, false
-
-testRelation(<)
-// CHECK-NEXT: false, true, false, false, true, false.
+expectEqual(testRelation(==), [true, false, false, false, false, true])
+expectEqual(testRelation(!=), [false, true, true, true, true, false])
+expectEqual(testRelation(<),  [false, true, false, false, true, false])
```

Comment:
Please move this into an `OptionalTest.test() {}`, there should be no top-level code.